### PR TITLE
Add jamming script

### PR DESCRIPTION
**Pull request by jonpas**
- Add jamming script (see header)
  - **Requires ACRE v2.8.0.1041+** ~~(not yet released)~~
- Add ACRE2 macros

Needs testing (simplified from Jack's version into single-function with parameters) and verification that parameters are applied properly. Probably also sane examples.

### CODE REVIEW

**Review by TyroneMF — APPROVED** [review]

**Comment by TyroneMF** [comment]
Will test in a mission before adding.

File: Scripts/fn_jam.sqf [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+_degradation params [
+    ["_strengthReduction", [], [0], [0, _countJammers]],
+    ["_decibelReduction", [], [0], [0, _countJammers]]
+];
+
+// Make global for PFH/EH use
+GVAR(activeJammers) = _jammers;
+
+// Remove jammer objects on destruction
+private _jammerClasses = _jammers arrayIntersect _jammers; // remove duplicates
```
Comment (jonpas) [inline Scripts/fn_jam.sqf:36]:
```suggestion
private _jammerClasses = _jammers apply {typeOf _x};
_jammerClasses = _jammers arrayIntersect _jammers; // remove duplicates
```

File: Scripts/fn_jam.sqf [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * Examples:
+ * [[jammer1, jammer2]] call TAC_Scripts_fnc_jam
+ * [[jammer1, jammer2], [[0.1, 0.5], [20, 60]]] call TAC_Scripts_fnc_jam
+ */
+
+params [["_jammers", [], [[]]], ["_degradation", [], [[]]]];
+private _countJammers = count _jammers;
+_degradation params [
+    ["_strengthReduction", [], [0], [0, _countJammers]],
+    ["_decibelReduction", [], [0], [0, _countJammers]]
```
Comment (jonpas) [inline Scripts/fn_jam.sqf:29]:
```suggestion
    ["_strengthReduction", [], [[]], [0, _countJammers]],
    ["_decibelReduction", [], [[]], [0, _countJammers]]
```

File: Scripts/fn_jam.sqf [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+];
```
Comment (jonpas) [inline Scripts/fn_jam.sqf:30]:
```suggestion
params [["_jammers", [], [[]]], ["_degradation", [], [[]]]];

if (count _jammers != count _degradation) exitWith {
    ERROR_WITH_TITLE("Invalid signal jamming parameters!", "Expected 0 or %1 degradation parameters for %1 jammers",count _jammers);
};

_degradation params [
    ["_strengthReduction", [], [[]], [2]],
    ["_decibelReduction", [], [[]], [2]]
];
```

File: Scripts/fn_jam.sqf [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        GVAR(activeJammers) = GVAR(activeJammers) - [_object];
+        publicVariable QGVAR(activeJammers);
+    }] call CBA_fnc_addClassEventHandler;
+} forEach _jammerClasses;
+
+// Register custom signal function
+[{
+    // ACRE signal processing
+    private _coreSignal = _this call ACREFUNC(sys_signal,getSignalCore);
+    _coreSignal params ["_f", "_mW", "_receiverClass", "_transmitterClass"];
```
Comment (jonpas) [inline Scripts/fn_jam.sqf:49]:
```suggestion
    _coreSignal params ["_Px", "_maxSignal"];
```

**Comment by TyroneMF** [comment]
Initial testing works, needs another one with values provided by someone who understands this sorcery.

**Comment by jonpas** [comment]
Moved to https://github.com/Theseus-Aegis/Mods/pull/542.

**Pull request closed without merging**